tests/core/test_lightning_optimizer.py

```diff
@@ -287,25 +287,30 @@ def step(self, closure=None):
 
 
 def test_lightning_optimizer_keeps_hooks(tmpdir):
-    class TestModel(BoringModel):
-        count_on_train_batch_start = 0
-        count_on_train_batch_end = 0
+    model = BoringModel()
+    optimizer = OptimizerWithHooks(model)
+    lightning_optimizer = LightningOptimizer(optimizer)
+    assert len(optimizer._fwd_handles) == 1
+    del lightning_optimizer
+    assert len(optimizer._fwd_handles) == 1
 
-        def configure_optimizers(self):
-            return OptimizerWithHooks(self)
 
-        def on_train_batch_start(self, batch: Any, batch_idx: int) -> None:
-            self.count_on_train_batch_start += 1
-            optimizer = self.optimizers(use_pl_optimizer=False)
-            assert len(optimizer._fwd_handles) == 1
+def test_init_optimizers_resets_lightning_optimizers(tmpdir):
+    """Test that the Trainer resets the `lightning_optimizers` list everytime new optimizers get initialized."""
 
-        def on_train_batch_end(self, outputs: Any, batch: Any, batch_idx: int) -> None:
-            self.count_on_train_batch_end += 1
-            del self.trainer._lightning_optimizers
```

Comment:
I don't think allowing `del lightning_optimizers` during fitting makes sense.

Refactored the test to not use `fit`